docs/ops/matrix/Einsum_7.md

```diff
@@ -0,0 +1,228 @@
+
+```
+A = [[1.0, 2.0, 3.0],
+     [4.0, 5.0, 6.0],
+     [7.0, 8.0, 9.0]]
+equation = "a...->..."
+output = [12.0, 15.0, 17.0]
+```
+
+Example 7 shows how *Einsum* operates with broadcasting two operands:
+
+```
+A = [[1.0, 2.0, 3.0],
+     [4.0, 5.0, 6.0],
+     [7.0, 8.0, 9.0]]
+B = [0.5]
+equation = "a...,...->a..."
+output = [[0.5, 1.0, 1.5],
+          [2.0, 2.5, 3.0],
+          [3.5, 4.0, 4.5]]
+```
+
+**NOTE**: Deep learning frameworks support implicit mode (a classical form of Einstein summation) where the equation does not have the output subscript 
+and has the following format: `<subscript for input1>, <subscript for input2>, ..., <subscript for inputn>`.
+The equation consists of only input subscripts for each operand.
+The output subscript can be recovered as a sequence of alphabetically sorted labels that are not repeated in the left-hand side of the equation.
+For example, `equation = "dbbc,ca"` in implicit mode is the same as `equation = "dbbc,ca->ad"`.
+The equation in implicit mode can set up only subset of Einstein summation conventions, so this mode has limitations. For example,
+`equation = "kii->i"` cannot be represented in implicit mode. *Einsum* operation only supports explicit mode.
+In case ellipsis label is in the left-hand side of the equation in implicit mode, the ellipsis comes first in the output subscript for the recovery.
+
+**Attributes**:
+
+* *equation*
+
+  * **Description**: it defines Einstein summation convention on input operands. The equation must be in explicit mode, meaning the output subscript must be in place.
+  * **Range of values**: the equation format is described above
+  * **Type**: string
+  * **Required**: *yes*
+
+**Inputs**:
+
+* **Multiple inputs**: Tensors of type *T* and different shapes.
+
+**Output**:
+
+*   **1**: Tensor of type *T* and shape is computed based on the output subscript of the equation.
+
+**Types**
+
+* *T*: any numeric type.
+
+**Examples**
+
+```xml
+<layer ... type="Einsum" version="opset7">
+    <data equation="ij,ij->i"/>
+    <input>
+        <port id="0">
+            <dim>2</dim>
+            <dim>64</dim>
+        </port>
+        <port id="0">
+            <dim>2</dim>
+            <dim>64</dim>
+        </port>
+    </input>
+    <output>
+        <port id="2">
+            <dim>2</dim>
+        </port>
+    </output>
+</layer>
+```
+
+```xml
+<layer ... type="Einsum" version="opset7">
+    <data equation="ab...,ac...,ade->...bc"/>
+    <input>
+        <port id="0">
+            <dim>2</dim>
+            <dim>3</dim>
+            <dim>4</dim>
+        </port>
+        <port id="1">
+            <dim>2</dim>
+            <dim>7</dim>
+            <dim>1</dim>
+        </port>
+        <port id="3">
+            <dim>2</dim>
+            <dim>4</dim>
+            <dim>7</dim>
+        </port>
+    </input>
+    <output>
+        <port id="4">
+            <dim>4</dim>
```

Comment:
I didn't get why this dimension is equal to 4. Can you explain the evaluation step by step?

Reply:
This logic was explained above in paragraph about ellipsis. In this case, we have two operands with broadcasted dimensions: the first operand with broadcasted dimension ([4]) and the second operand with broadcasted dimension ([1]). They are broadcasted to the common shape [4]